### PR TITLE
chore(botocore): allow flaky test to fail

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
https://github.com/DataDog/dd-trace-py/pull/3178 introduced a flaky
test, until it can be fixed we'll allow it to fail.

### CODE REVIEW

File: tests/contrib/botocore/test.py [hunk 1]
```diff
@@ -1013,6 +1014,7 @@ def test_sns_send_message_trace_injection_with_no_message_attributes(self):
 
     @mock_sns
     @mock_sqs
+    @pytest.mark.xfail(strict=False)  # FIXME: flaky test
```
Comment (brettlangdon) [inline tests/contrib/botocore/test.py:1017]: how does `xfail(strict=False)` work?

we still run the best, but we just don't care about the results?

- No `xfail()`: it must pass
- `xfail()`: it must fail
- `xfail(strict=False)`: it is run, but the result doesn't matter
- `xskip()`: don't run the test

?

Comment (Kyle-Verhoog) [reply tests/contrib/botocore/test.py:1017]: https://docs.pytest.org/en/6.2.x/flaky.html#xfail-strict

Yeah run it and report the failure but don't fail the run.

**Comment by P403n1x87** [comment]
Would the test pass if re-run? Can we perhaps benefit from the [flaky](https://pypi.org/project/flaky/) plugin here?

**Review by brettlangdon — APPROVED** [review]

**Comment by brettlangdon** [comment]
> Would the test pass if re-run? Can we perhaps benefit from the [flaky](https://pypi.org/project/flaky/) plugin here?

I don't know if it matters much either way. I think the difficult part is actually knowing that there is a test temporarily marked as flaky (either re-run, xfail, xfail(strict=True), xskip, etc) and then to come back to it.

Maybe we only mark for re-run tests which there is no way to make them not flaky but which can only succeed if run multiple times .... ? (/me is trying to invent a heuristic)

**Review by majorgreys — APPROVED** [review]

**Pull request merged**